mungegithub/pulls/needs_rebase.go

```diff
@@ -26,31 +24,47 @@ import (
 )
 
 type RemoveNeedsRebaseMunger struct{}
+type AddNeedsRebaseMunger struct{}
+
+const needsRebase = "needs-rebase"
 
 func init() {
 	RegisterMungerOrDie(RemoveNeedsRebaseMunger{})
+	RegisterMungerOrDie(AddNeedsRebaseMunger{})
 }
 
 func (RemoveNeedsRebaseMunger) Name() string { return "remove-needs-rebase" }
 
 func (RemoveNeedsRebaseMunger) MungePullRequest(client *github.Client, pr *github.PullRequest, issue *github.Issue, commits []github.RepositoryCommit, events []github.IssueEvent, opts opts.MungeOptions) {
-	if !HasLabel(issue.Labels, "needs-rebase") {
+	if !HasLabel(issue.Labels, needsRebase) {
 		return
 	}
 	if pr.Mergeable == nil {
-		// If mergeable is nil, github is likely re-calculating mergeability, wait a bit and reload the PR.
-		time.Sleep(2 * time.Second)
-		var err error
-		if pr, _, err = client.PullRequests.Get(opts.Org, opts.Project, *pr.Number); err != nil {
-			glog.Errorf("Failed to resync project: %d", *pr.Number)
-			return
-		}
+		glog.Infof("Skipping %d since mergeable is nil", *pr.Number)
 	}
 	if pr.Mergeable != nil && *pr.Mergeable {
 		if opts.Dryrun {
 			glog.Infof("Would have removed needs-rebase for %d", *pr.Number)
 		} else {
-			client.Issues.RemoveLabelForIssue(opts.Org, opts.Project, *pr.Number, "needs-rebase")
+			client.Issues.RemoveLabelForIssue(opts.Org, opts.Project, *pr.Number, needsRebase)
+		}
+	}
```

Comment:
Done.  (I had thought it would be an ugly mess of if/else conditions, but it's not that bad...)